src/main/java/org/jabref/logic/xmp/XMPUtilReader.java

```diff
@@ -0,0 +1,124 @@
+package org.jabref.logic.xmp;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+
+import org.jabref.model.entry.BibEntry;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.apache.pdfbox.pdmodel.PDDocumentCatalog;
+import org.apache.pdfbox.pdmodel.PDDocumentInformation;
+import org.apache.pdfbox.pdmodel.common.PDMetadata;
+import org.apache.xmpbox.XMPMetadata;
+import org.apache.xmpbox.schema.DublinCoreSchema;
+
+public class XMPUtilReader {
+
+    private XMPUtilReader() {
+    }
+
+    /**
+     * Will read the XMPMetadata from the given pdf file, closing the file
+     * afterwards.
+     *
+     * @param path The path to read the XMPMetadata from.
+     * @return The XMPMetadata object found in the file
+     */
+    public static Optional<XMPMetadata> readRawXMP(Path path) throws IOException {
+        try (PDDocument document = XMPUtilReader.loadWithAutomaticDecryption(path)) {
+            return XMPUtilReader.getXMPMetadata(document);
+        }
+    }
+
+    /**
+     * Convenience method for readXMP(File).
+     *
+     * @param filename The filename from which to open the file.
+     * @return BibtexEntryies found in the PDF or an empty list
+     */
+    public static List<BibEntry> readXMP(String filename, XMPPreferences xmpPreferences) throws IOException {
+        return XMPUtilReader.readXMP(Paths.get(filename), xmpPreferences);
+    }
+
+    /**
+     * Try to read the given BibTexEntry from the XMP-stream of the given
+     * inputstream containing a PDF-file.
+     *
+     * Only supports Dublin Core as a metadata format.
+     *
+     * @param path The path to read from.
+     * @return list of BibEntries retrieved from the stream. May be empty, but never null
+     * @throws IOException Throws an IOException if the file cannot be read, so the user than remove a lock or cancel
+     *                     the operation.
+     */
+    public static List<BibEntry> readXMP(Path path, XMPPreferences xmpPreferences)
+            throws IOException {
+
+        List<BibEntry> result = new LinkedList<>();
+
+        try (PDDocument document = loadWithAutomaticDecryption(path)) {
+            Optional<XMPMetadata> meta = XMPUtilReader.getXMPMetadata(document);
+
+            if (meta.isPresent()) {
```

Comment:
You can use chained maps on the optionals for each step:
meta.map(m->m.getDublinCore().map(m->new DublinCore...).map(entry->extractBibEntry).ifPresent(result::add)

Reply:
I find the current state more readable than the chained map operations... Do you have another suggestion?

Reply:
The chanining has the improvement that you don't have to care about nulls or empty optionals inside. The code at the end is only executed if none of the values is null or empty.

Did not see your question. Theoretically it could be that someone has CrossRef linked bibentries, (e.g. a bib entry with a book and another one with a chapter referring the book).
 but that's really and edge case. I think for most uses cases  one entry is enough. And if the code doesn't support it, then drop it.  let's see what @koppor  has to say